tests/nodetree/test_node_unit.py

```diff
@@ -0,0 +1,285 @@
+            assert keys <= set(result_node.__dir__())
+            for k in keys:
+                pythonify_mock.assert_any_call(k)
+
+    @pytest.mark.parametrize(('next_segment', 'subtree_paths','expected'),
+                             [('a', set(), False),
+                              ('a', {'a', 'b'}, True),
+                              ('0', {'0'}, True),
+                              ('c', {'a','c','e','f'}, True),
+                              ('d', {'a', 'c', 'e', 'f'}, False)
+                              ])
+    def test_contains_plain_value(self, next_segment, subtree_paths, expected):
+        node = MockResultNode(())
+        node._subtree_paths = subtree_paths
+
+        with patch('labone.nodetree.node.normalize_path_segment',
+                   side_effect=lambda x: x) as normal_patch:
+            decision = node.__contains__(next_segment)
+            normal_patch.assert_called_once_with(next_segment)
+            assert decision == expected
+
+    @pytest.mark.parametrize(('next_segment'),
+                             ['a',
+                              'b',
+                              ])
+    def test_contains_node(self, next_segment):
+        arg = MockResultNode((next_segment,))
+
+        with patch('labone.nodetree.node.ResultNode.is_child_node') as child_patch:
+            MockResultNode(()).__contains__(arg)
+            child_patch.assert_called_once_with(arg)
+
+    @pytest.mark.parametrize(('next_segment'),
+                             ['a',
+                              'b',
+                              ])
+    def test_contains_annotated_value(self, next_segment):
+        path = join_path(next_segment)
+
+        with patch('labone.nodetree.node.ResultNode.is_child_node') as child_patch:
+            with patch('labone.nodetree.node.split_path',
+                       return_value=next_segment) as split_patch:
+                MockResultNode(()).__contains__(AnnotatedValue(path=path, value=0))
+                split_patch.assert_called_once_with(path)
+                child_patch.assert_called_once_with(next_segment)
+
+    def test_call(self):
+        with pytest.raises(LabOneInappropriateNodeTypeError):
+            MockResultNode(()).__call__()
+
+    def test_str(self, result_node):
+        assert isinstance(str(result_node), str)
+
+    def test_repr(self, result_node):
+        assert isinstance(repr(result_node), str)
+
+    @pytest.mark.parametrize(('path', 'expected'),
+                             [('/a', {1,2,3,4,5}),
+                              ('/b', set()),
+                              ('/a/b', {2,3,4,5}),
+                              ('/a/b/d', {4,5}),
+                              ('/a/b/d/e', {5}),
+                              ('/a/b/c', {3}),
+                              ('/a/b/e', set())
+                              ])
+    def test_results(self, path, expected):
+        value_structure = {'/a': 1, '/a/b': 2, '/a/b/c': 3, '/a/b/d': 4, '/a/b/d/e':5}
```

Comment:
suggestion: I would change this to something that would actually work 😇 

```suggestion
        value_structure = {'/a/b': 1, '/a/c/d': 2, '/a/c/e': 3, '/a/c/f/g':4, '/a/c/f/h':5}
```